src/ATen/native/xpu/mkl/BatchLinearAlgebra.cpp

```diff
@@ -0,0 +1,352 @@
+    const Tensor& self,
+    bool some,
+    bool compute_uv) {
+  auto infos_tensor = at::zeros(
+      native::batchCount(self),
+      self.options().dtype(kInt).device(DeviceType::CPU));
+  std::vector<int32_t> infos(native::batchCount(self), 0);
+
+  char jobz = compute_uv ? (some ? 'S' : 'A') : 'N';
+
+  Tensor U_working_copy, S_working_copy, VT_working_copy;
+  std::tie(U_working_copy, S_working_copy, VT_working_copy) =
+      impl::_create_U_S_VT(self, some, compute_uv);
+
+  if (self.numel() > 0) {
+    auto self_working_copy = native::cloneBatchedColumnMajor(self);
+    auto& queue = at::xpu::getCurrentSYCLQueue();
+    auto self_stride = at::native::matrixStride(self_working_copy);
+    auto U_stride = compute_uv ? at::native::matrixStride(U_working_copy) : 1;
+    auto S_stride = S_working_copy.size(-1);
+    auto VT_stride = compute_uv ? at::native::matrixStride(VT_working_copy) : 1;
+    auto batchsize = at::native::batchCount(self_working_copy);
+
+    auto m = self_working_copy.size(-2);
+    auto n = self_working_copy.size(-1);
+    int64_t lda = self_working_copy.stride(-1);
+    int64_t ldu = compute_uv ? U_working_copy.stride(-1) : 1;
+    int64_t ldvt = compute_uv ? VT_working_copy.stride(-1) : 1;
+    AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES(self.scalar_type(), "svd_xpu", [&] {
+      using value_t = typename c10::scalar_value_type<scalar_t>::type;
+      impl::apply_svd<scalar_t, value_t>(
+          queue,
+          self_working_copy.data_ptr<scalar_t>(),
+          lda,
+          self_stride,
+          batchsize,
+          m,
+          n,
+          self.options(),
+          compute_uv ? U_working_copy.data_ptr<scalar_t>() : nullptr,
+          ldu,
+          U_stride,
+          S_working_copy.data_ptr<value_t>(),
+          S_stride,
+          compute_uv ? VT_working_copy.data_ptr<scalar_t>() : nullptr,
+          ldvt,
+          VT_stride,
+          jobz);
+    });
+
+    std::copy(
+        infos.begin(), infos.end(), infos_tensor.template data_ptr<int32_t>());
+    at::_linalg_check_errors(infos_tensor, "svd_xpu", self.dim() == 2);
+
+    if (!compute_uv) {
+      VT_working_copy.zero_();
+      U_working_copy.zero_();
+    }
+  } else {
+    U_working_copy.zero_();
+    VT_working_copy.zero_();
+  }
+
+  return std::make_tuple(U_working_copy, S_working_copy, VT_working_copy);
+}
+
+static void svd_resize_and_copy(
+    const char* name,
+    const Tensor& src,
+    const Tensor& dst) {
+  TORCH_CHECK(
+      src.device() == dst.device(),
+      "svd output tensor ",
+      name,
+      " is on the wrong device: expected ",
+      src.device(),
+      " got ",
+      dst.device());
+  at::native::resize_output(dst, src.sizes());
+  dst.copy_(src);
+}
+
+void svd_mkl(
+    const Tensor& A,
+    const bool full_matrices,
+    const bool compute_uv,
+    const c10::optional<c10::string_view>& driver,
+    const Tensor& U,
+    const Tensor& S,
+    const Tensor& Vh,
+    const Tensor& info) {
```

Comment:
Can you align this op implementation with `svd_stub` logic in stock pytorch. Previous code in ipex is based on a legacy impl in Pytorch 1.x. For our current upstream solution, those copy and resize etc. should already be covered by common code in stock pytorch now. please check.

Reply:
I tried to remove those resize and copy but the current implementation will be broken. My suggestion is to ensure functionality on XPU first and refine these code in another PR.

Reply:
@CuiYifeng , any justification regarding these operations that we need to ensure the functionality first? Any models that depend on these operations? If not, why don't we polish this PR to make it good enough?

Reply:
SVD was requested by Argonne. Get your point.